pkg/apis/workflow/v1alpha1/common.go

```diff
@@ -27,3 +28,18 @@ type TemplateReferenceHolder interface {
 	GetTemplateName() string
 	GetTemplateRef() *TemplateRef
 }
+
+// SubmitOpts are workflow submission options
+type SubmitOpts struct {
+	Name           string                 `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`                      // --name
```

Comment:
Can you move the comment above the line like this:

```
// --name
Name string `json...
```

I know you didn't add them, but them being on the same line will cause the swagger files to think the comments are for the field below. Take a look at a swagger file to see what I mean.